app/helpers/metrics_helper.rb

```diff
@@ -12,51 +12,73 @@ def sum_rollups(rollups)
     rollups.sum(&:count)
   end
 
+  def podcast_main_card_options
+    %i[downloads episodes uniques dropdays]
+  end
+
+  def episode_main_card_options
+    %i[downloads]
+  end
+
   def interval_options
     Rollups::HourlyDownload::INTERVALS.map { |i| [I18n.t(".helpers.label.metrics.interval.#{i.downcase}"), i] }
   end
 
-  def date_range_options(episode = nil)
+  def date_preset_options(episode = nil)
     if episode
-      episode_date_presets(episode)
+      episode_date_preset_options
     else
-      podcast_date_presets
+      podcast_date_preset_options
     end
   end
 
-  def podcast_date_presets
-    metrics_date_presets(Rollups::HourlyDownload::PODCAST_DATE_PRESETS)
+  def podcast_date_preset_options
+    Rollups::HourlyDownload::PODCAST_DATE_PRESETS.map do |preset|
+      [label_for_date_preset(preset), preset.to_s]
+    end
   end
 
-  def episode_date_presets(episode)
-    metrics_date_presets(Rollups::HourlyDownload::EPISODE_DATE_PRESETS, episode)
+  def episode_date_preset_options
+    Rollups::HourlyDownload::EPISODE_DATE_PRESETS.map do |preset|
+      [label_for_date_preset(preset), preset.to_s]
+    end
   end
 
-  def metrics_date_presets(options, episode = nil)
-    options.map do |opt|
-      count, interval, type = opt.to_s.split("_")
+  def dates_from_preset(preset, episode = nil)
+    date_start = if date_start_from_preset(preset)
+      date_start_from_preset(preset)
+    elsif episode
+      episode.first_publish_utc_date
+    else
+      Date.utc_today - 1.day
+    end
 
-      date_start = if type == "last"
-        count.to_i.send(interval).ago.utc_date
-      elsif type == "previous"
-        (Date.utc_today - count.to_i.send(interval)).send(:"beginning_of_#{interval.singularize}")
-      elsif count == "date"
-        Date.utc_today.send(:"beginning_of_#{interval.singularize}")
-      elsif episode
-        episode.first_publish_utc_date
-      else
-        Date.utc_today - 1.day
-      end
+    date_end = date_end_from_preset(preset, date_start)
+
+    [date_start, guard_date_end(date_end)]
+  end
 
-      date_end = if type == "drop"
-        date_start + count.to_i.send(interval)
-      elsif type == "previous"
-        (date_start + (count.to_i - 1).send(interval)).send(:"end_of_#{interval.singularize}")
-      else
-        Date.utc_today
-      end
+  def date_start_from_preset(preset)
+    type, count, interval = preset.to_s.split("_")
 
-      [I18n.t(".helpers.label.metrics.date_presets.#{opt}"), date_preset(date_start, date_end)]
+    if type == "last"
+      count.to_i.send(interval).ago.utc_date
+    elsif type == "previous"
+      (Date.utc_today - count.to_i.send(interval)).send(:"beginning_of_#{interval.singularize}")
+    elsif type == "todate"
+      Date.utc_today.send(:"beginning_of_#{interval.singularize}")
+    end
+  end
+
+  def date_end_from_preset(preset, date_start)
```

Comment:
`date_end` sometimes depends on what the `date_start` is.

example of type `drop` is 30 Days Since Drop. only applies to Episode level Metrics, since Podcast level Metrics has its own `Dropdays` tab. `date_start` in this case uses the default for an episode which is `episode.first_publish_utc_date`, and then counts from there.

example of type `previous` is Previous 3 Months. if we are in October, `date_start` would start at the beginning of July. this condition then adds (3 - 1) months to bring us to September, and sets the value to the end of month (Sep 30).